js/binance.js

```diff
@@ -5246,4 +5259,68 @@ module.exports = class binance extends Exchange {
         }
         return result;
     }
+
+    async fetchOpenInterestHistory (symbol, timeframe = '5m', since = undefined, limit = undefined, params = {}) {
+        if (timeframe === '1m') {
+            throw new BadRequest (this.id + 'fetchOpenInterestHistory cannot use the 1m timeframe');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'period': this.timeframes[timeframe],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const symbolKey = market['future'] ? 'symbol' : 'pair';
+        request[symbolKey] = market['id'];
+        if (market['delivery']) {
+            request['contractType'] = this.safeString (params, 'contractType', 'CURRENT_QUARTER');
+        }
+        if (since !== undefined) {
+            request['startTime'] = since;
+        }
+        const till = this.safeInteger (params, 'till'); // unified in milliseconds
+        const endTime = this.safeString (params, 'endTime', till); // exchange-specific in milliseconds
+        params = this.omit (params, [ 'endTime', 'till' ]);
+        if (endTime) {
+            request['endTime'] = endTime;
+        } else if (since) {
+            if (limit === undefined) {
+                limit = 30; // Exchange default
+            }
+            request['endTime'] = Precise.stringAdd (since.toString (), Precise.stringMul (this.milli[timeframe], limit.toString ()));
```

Comment:
We should not really do the string-based math upon sending the request here.
Also, we don't need the `milli` thing. We have `this.parseTimeframe` for this matter, that returns the duration in seconds. Let's rewrite it as follows:

```JavaScript
const duration = this.parseTimeframe (timeframe);
request['endTime'] = this.sum (since, duration * limit * 1000);
```